### PR TITLE
Add more details for cache control

### PR DESCRIPTION
**Pull request by mrkschan**
Reference - https://www.nginx.com/blog/nginx-caching-guide/

### CODE REVIEW

**Comment by mrkschan** [comment]
Does anyone has idea on when Nginx revalidate `ETag` in a response?

**Comment by mrkschan** [comment]
Found this :) https://trac.nginx.org/nginx/timeline?from=2014-07-01T11%3A44%3A24Z&precision=second

File: reference/cache.md [hunk 1]
```diff
@@ -69,6 +68,16 @@ And the following routes are **not** cached:
...
 
+## Cache behavior
+
+Cache is only applied to `GET` and `HEAD` requests.
+Responses with `Cache-Control` header set to `Private`, `No-Cache`,
+or `No-Store` are not cached. Also, this applies to responses with
+`Set-Cookie` header being set.
+
+Cache would also respect the `Vary` header in the response.
+Multiple versions of the cache will be created if `Vary` header is set.
```
Comment (ralt) [inline reference/cache.md]: Between the `Vary` header and the cache key defined in the `routes.yaml`, how does it play together?

Comment (mrkschan) [inline reference/cache.md]: `Vary` is similar to cache key. See https://www.nginx.com/blog/nginx-plus-r5-released/.

Comment (ralt) [inline reference/cache.md]: Maybe this should be properly documented? I won't be the only one asking that.

Comment (mrkschan) [inline reference/cache.md]: From http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_valid

> If the header includes the “Vary” field with the special value “*”, such a response will not be cached (1.7.7). If the header includes the “Vary” field with another value, such a response will be cached taking into account the corresponding request header fields (1.7.7). 

First we need a understanding :)

Comment (mrkschan) [inline reference/cache.md]: Relevant code block? https://github.com/nginx/nginx/blob/bf838dc9dfca4e80b883c2e659c65f90fc1526a3/src/http/ngx_http_file_cache.c#L525

Comment (mrkschan) [inline reference/cache.md]: The cache data structure has `vary` and `varient` :) https://github.com/nginx/nginx/blob/c3aed0a23392a509f64b740064f5f6633e8c89d8/src/http/ngx_http_cache.h#L64

Comment (mrkschan) [inline reference/cache.md]: Example with cache key and vary header playing together - http://marc.info/?l=nginx&m=144432816821289&w=2

File: reference/cache.md [hunk 2]
```diff
@@ -90,10 +83,41 @@ The default value for these keys are the following:
 ```yaml
 cache:
   enabled: true
-  headers: ["Accept-Language", "Accept"]
+  headers: []
   cookies: ["*"]
 ```
 
+`Vary` header in the response is also respected. Multiple copies under the
```
Comment (Crell) [inline reference/cache.md]: Missing "The" at the start of the sentence.

File: reference/cache.md [hunk 3]
```diff
@@ -90,10 +83,41 @@ The default value for these keys are the following:
...
+same cache key would be created according to the value of this header.
+
+## Cache behavior
+
+Cache is only applied to `GET` and `HEAD` requests.
+Responses with `Cache-Control` header set to `Private`, `No-Cache`,
+or `No-Store` are not cached. Also, this applies to responses with
+`Set-Cookie` header being set.
```
Comment (Crell) [inline reference/cache.md]: Rephrase: "Responses with the `Set-Cookie` header set are also not cached."

File: reference/cache.md (hunk 1 above)
Comment (ralt) [inline reference/cache.md]: Either way, we need to make that clear in the documentation.

**Comment by mrkschan** [comment]
@ralt @Crell updated.

**Pull request merged**